### PR TITLE
Add support for locking with POST

### PR DESCRIPTION
**Pull request by derek63**

### CODE REVIEW

File: core/src/main/java/com/redhat/lightblue/client/LightblueClient.java [hunk 1]
```diff
@@ -12,6 +11,14 @@
 
     Locking getLocking(String domain);
 
+    /**
+     * This method allows
+     * @param domain
+     * @param usePost
+     * @return
+     */
+    Locking getLocking(String domain, Boolean usePost);
```
Comment (bserdar) [inline core/src/main/java/com/redhat/lightblue/client/LightblueClient.java]: I don't think there's a point to preserving the old lock API calls in the client. One ctor should be enough, using only POST.

File: http/src/main/java/com/redhat/lightblue/client/http/LightblueHttpClient.java [hunk 2]
```diff
@@ -63,34 +106,39 @@ public LockingRequest(String domain, String callerId, String resourceId, Long tt
...
-            b.append(uri);
-            return b.toString();
+        private String getRestURIWithPost() {
+            return "lock/";
         }
     }
 
     private final class LockingImpl extends Locking {
 
+        Boolean usePost = false;
```
Comment (bserdar) [inline http/src/main/java/com/redhat/lightblue/client/http/LightblueHttpClient.java]: Same here: no need for usePost. Client can always use POST.

File: http/src/main/java/com/redhat/lightblue/client/http/LightblueHttpClient.java [hunk 3]
```diff
@@ -76,21 +80,19 @@ public String getRestURI(String baseServiceURI) {
             if (!baseServiceURI.endsWith("/")) {
                 b.append('/');
             }
-            b.append(uri);
+            b.append("lock/");
```
Comment (bserdar) [inline http/src/main/java/com/redhat/lightblue/client/http/LightblueHttpClient.java]: trailing /?

Comment (derek63) [inline http/src/main/java/com/redhat/lightblue/client/http/LightblueHttpClient.java]: I don't think its needed, but saw it elsewhere so probably just being consistent

Comment (derek63) [inline http/src/main/java/com/redhat/lightblue/client/http/LightblueHttpClient.java]: Works with no / too, I will remove it

**Pull request merged**